### PR TITLE
Added safe_add sql function

### PR DESCRIPTION
**Pull request by Deegiimurun**
Is this a user-visible change (yes/no): no

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresNumericTests.java [hunk 1]
```diff
@@ -547,6 +547,19 @@ public void testAdd() {
         this.testTwoViews(intermediate, last);
     }
 
+
+    @Test
+    public void testSafeAdd() {
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresNumericTests.java:552]: This file only contains tests from the postgres testsuite, as described in the first comment.
So I would move these tests to a separate file in sql, perhaps in the `simple` directory.
I would also write some new tests which overflow, that's the point of this function.
In fact, safe_add is a bigquery SQL extension, so you probably won't find tests for it in Postgres or mysql.

File: sql-to-dbsp-compiler/lib/sqllib/src/operators.rs [hunk 2]
```diff
@@ -313,3 +313,12 @@ pub fn div_dN_dN(left: Option<F64>, right: Option<F64>) -> Option<F64> {
 pub fn plus_u_u(left: usize, right: usize) -> usize {
     left + right
 }
+
+#[inline(always)]
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/lib/sqllib/src/operators.rs:317]: you need many more functions here, for all types that are supported by safe_add, following the example of `plus`.
you probably also need some calls to the macro `for_all_int_operators!`

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/lib/sqllib/src/operators.rs:317]: The documentation is also missing.
